sw/device/silicon_creator/lib/boot_data.c

```diff
@@ -417,31 +381,25 @@ rom_error_t boot_data_write(const boot_data_t *boot_data) {
     RETURN_IF_ERROR(boot_data_digest_compute(&new_entry, &new_entry.digest));
     if (active_page.has_empty_entry == kHardenedBoolTrue) {
       RETURN_IF_ERROR(boot_data_entry_write(
-          active_page.base_addr, active_page.first_empty_index, &new_entry));
+          active_page.page, active_page.first_empty_index, &new_entry));
     } else {
-      uint32_t new_page_base = active_page.base_addr == kBootDataPage0Base
-                                   ? kBootDataPage1Base
-                                   : kBootDataPage0Base;
-      // TODO(#8777): Update error handling after switching to silicon_creator
-      // driver.
-      if (flash_page_erase(new_page_base, kInfoPartition) != 0) {
-        return kErrorBootDataFlash;
-      }
-      RETURN_IF_ERROR(boot_data_entry_write(new_page_base, 0, &new_entry));
+      flash_ctrl_info_page_t new_page =
```

Comment:
Might be worth adding a brief comment as to what is going on here.

Reply:
Done.